src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/ActionMetric.kt

```diff
@@ -77,17 +77,29 @@ data class ActionMetric @Deprecated("Use ActionMetric.Builder instead.") constru
 
     class Builder(
         private val label: String,
-        private val result: ActionResult,
-        private val duration: Duration,
+        private var result: ActionResult,
+        private var duration: Duration,
         private val start: Instant
     ) {
         private var observation: JsonObject? = null
         private var drilldown: RecordedPerformanceEntries? = null
         private var virtualUser: UUID = UUID.randomUUID()
 
+
+        constructor(label: String, virtualUser: UUID, start: Instant) : this(
```

Comment:
Ad hoc constructors in Builders are suspicious. Builders hold defaults of all optional fields and enforce passing the required fields. Every other permutation can be reduced to calling the required constructor and mutators.
Builders only need the required-params (no good defaults) constructor and a copying constructor.

- [ ] avoid ad hoc builder constructors